test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -513,6 +555,69 @@ def test_basic_save_and_load_state_dict(
 
             self._compare_models(model, model_new, self.assertEqual, check_fp16=fp16)
 
+    @skip_if_lt_x_gpu(2)
+    @parametrize("state_dict_type", _SUPPORTED_STATE_DICT_IMPLS)
+    @parametrize(
+        "cpu_offload",
+        [CPUOffload(offload_params=True), CPUOffload(offload_params=False)],
+    )
+    @parametrize("mixed_precision", [True, False])
+    @parametrize("state_dict_rank0_and_offload", [True, False])
+    def test_buffers_save_and_load_state_dict(
+        self,
+        state_dict_type: StateDictType,
+        cpu_offload: bool,
+        mixed_precision: bool,
+        state_dict_rank0_and_offload: bool,
+    ):
+        """
+        Tests that we can save a state_dict and load it for modules with persistent buffers, including
+        in the context of non-default mixed precision, different ``state_dict_type`` s and CPU offloading.
+        """
+        if state_dict_rank0_and_offload and state_dict_type != "state_dict":
+            return  # not supported
```

Comment:
no need to fix in this PR, but this sort of pattern is not ideal from a TTS perspective, i.e. we still spawn N processes, initialize NCCL, etc only to return and tear everything down, paying unnecessary overhead.

Reply:
Agreed! Note, this (anti)pattern is used in `test_state_dict_with_manual_ac_wrapper` and `test_basic_save_and_load_state_dict` as well but those similar TTS issues can be addressed in the context of the refactoring options @awgu outlines below (either of which seem useful and worth doing!).

Reply:
One easy way to fix this anti pattern is to consolidate the usage of StateDictType and StateDictConfig. This worth another PR to fix. Thanks for pointing out the issue.